provider/mediaconvert/mediaconvert_test.go

```diff
@@ -0,0 +1,643 @@
+				p.Video.ProfileLevel = "9001"
+				return p
+			},
+			wantErrMsg: `generating video preset: h264 level "9001" is not supported with mediaconvert`,
+		},
+		{
+			name: "bad video width returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Video.Width = "s"
+				return p
+			},
+			wantErrMsg: `generating video preset: parsing video width "s" to int64: strconv.ParseInt: parsing "s": invalid syntax`,
+		},
+		{
+			name: "bad video height returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Video.Height = "h"
+				return p
+			},
+			wantErrMsg: `generating video preset: parsing video height "h" to int64: strconv.ParseInt: parsing "h": invalid syntax`,
+		},
+		{
+			name: "bad video bitrate returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Video.Bitrate = "bitrate"
+				return p
+			},
+			wantErrMsg: `generating video preset: parsing video bitrate "bitrate" to int64: strconv.ParseInt: parsing "bitrate": invalid syntax`,
+		},
+		{
+			name: "bad video gop size returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Video.GopSize = "gop"
+				return p
+			},
+			wantErrMsg: `generating video preset: parsing gop size "gop" to int64: strconv.ParseFloat: parsing "gop": invalid syntax`,
+		},
+		{
+			name: "unrecognized rate control mode returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.RateControl = "not supported"
+				return p
+			},
+			wantErrMsg: `generating video preset: rate control mode "not supported" is not supported with mediaconvert`,
+		},
+		{
+			name: "unrecognized interlace modes return an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Video.InterlaceMode = "unsupported mode"
+				return p
+			},
+			wantErrMsg: `generating video preset: h264 interlace mode "unsupported mode" is not supported with mediaconvert`,
+		},
+		{
+			name: "unrecognized audio bitrate returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Audio.Bitrate = "aud_bitrate"
+				return p
+			},
+			wantErrMsg: `generating audio preset: parsing audio bitrate "aud_bitrate" to int64: strconv.ParseInt: parsing "aud_bitrate": invalid syntax`,
+		},
+		{
+			name: "unrecognized audio codec returns an error",
+			presetModifier: func(p db.Preset) db.Preset {
+				p.Audio.Codec = "aab"
+				return p
+			},
+			wantErrMsg: `generating audio preset: audio codec "aab" is not yet supported with mediaconvert`,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			client := &testMediaConvertClient{t: t}
+			p := &mcProvider{client: client}
+			_, err := p.CreatePreset(tt.presetModifier(defaultPreset))
+			if err != nil && tt.wantErrMsg != err.Error() {
+				t.Errorf("mcProvider.CreatePreset() error = %v, wantErr %q", err, tt.wantErrMsg)
+				return
+			}
+
+			if tt.assertion != nil {
+				tt.assertion(client.createPresetCalledWith, t)
```

Comment:
This is an interesting pattern!

Reply:
I was looking for a handy way to test atomic aspects of the mapping, this seemed to work :)